geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/uuid/Z3FeatureIdGenerator.scala

```diff
@@ -75,7 +77,13 @@ object Z3UuidGenerator extends RandomLsbUuidGenerator {
     createUuid(geom.safeCentroid(), time, period)
   }
 
-  def createUuid(pt: Point, time: Long, period: TimePeriod) = {
+  def createUuid(pt: Point, time: Long, period: TimePeriod): UUID = {
+    // handle missing geometry
+    if (pt == null) {
+      logger.error("Cannot generate a meaningful UUID for a <NULL> geometry; returning a random UUID instead.")
```

Comment:
hmm, well I think it will fail further on when trying to [create mutations](https://github.com/locationtech/geomesa/blob/master/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/z3/Z3WritableIndex.scala#L58-L59). However, in a generic sense having the default id generator work without a geometry does seem appropriate (e.g. for a geometry-less schema).